platform/default/default_file_source.cpp

```diff
@@ -132,7 +123,7 @@ class DefaultFileSource::Impl {
 
     OfflineDatabase offlineDatabase;
     OnlineFileSource onlineFileSource;
-    std::unordered_map<AsyncRequest*, std::unique_ptr<Task>> tasks;
+    std::unordered_map<AsyncRequest*, std::unique_ptr<AsyncRequest>> tasks;
```

Comment:
Why is this an `unordered_map`? Pointers can be compared very fast and have their own well-defined order already, so creating a hash of a pointer just adds a level of indirection.